native/src/navigation/navigateToDeepLink.ts

```diff
@@ -43,63 +43,70 @@ const navigateToDeepLink = async <T extends RoutesType>(
     navigation.replace(INTRO_ROUTE, {
       deepLink: url
     })
-  } else {
-    const { pathname } = new Url(url)
-    const routeParser = new InternalPathnameParser(pathname, language, fixedCity)
-    const routeInformation = routeParser.route()
+    return
+  }
 
-    const routeInformationCityCode =
-      routeInformation &&
-      routeInformation.route !== LANDING_ROUTE &&
-      routeInformation.route !== JPAL_TRACKING_ROUTE &&
-      routeInformation.route !== CITY_NOT_COOPERATING_ROUTE
-        ? routeInformation.cityCode
-        : null
-    const routeInformationLanguageCode =
-      routeInformation && routeInformation.route !== LANDING_ROUTE && routeInformation.route !== JPAL_TRACKING_ROUTE
-        ? routeInformation.languageCode
-        : null
-    // Don't overwrite already selected city
-    const selectedCityCode = fixedCity || selectedCity || routeInformationCityCode || null
-    const languageCode = routeInformationLanguageCode || language
+  const { pathname } = new Url(url)
+  const routeInformation = new InternalPathnameParser(pathname, language, fixedCity).route()
 
-    if (selectedCityCode && languageCode) {
-      // Reset the currently opened screens to just the dashboard of the city and language
-      // This is necessary to prevent undefined behaviour for city content routes upon e.g. back navigation
-      navigateToCategory({
-        dispatch,
-        navigation,
-        cityCode: selectedCityCode,
-        languageCode,
-        routeName: DASHBOARD_ROUTE,
-        cityContentPath: createCityContentPath({
-          cityCode: selectedCityCode,
-          languageCode
-        }),
-        forceRefresh: false,
-        resetNavigation: true
-      })
-    } else {
-      navigation.replace(LANDING_ROUTE)
+  if (routeInformation?.route === JPAL_TRACKING_ROUTE && buildConfig().featureFlags.jpalTracking) {
```

Comment:
Only added this, rest is just indentation because else clause was removed by return in if.